### ISSUE TITLE
Add lacking creatures stats

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#152]
As discussed here: http://forum.freegamedev.net/viewtopic.php?p=59460#p59460

Once issue #151 is done, the creature's following stats will have to be added:
- [x] **XP table** 
  --> 30 values per creatures. Note that the creature's XP should go back to 0 at each level up as it isn't the case ATM. The XP to reach for next level should be displayed in the info window.
  (XP needed to level up, we should have 3 tables, weak creature, normal creature, strong creature; so weak monsters level up faster and stronger ones slower)
  The different XP tables should be found in the spreadsheet file. See #153 

**Stats Added:**
- [x] **Base HP** (at level 1)
- [x] **HP gain per level**
- [x] **Base physical attack**
- [x] **Physical attack gain / level**
- [x] **Base magical attack**
- [x] **Magical attack gain / level**
- [x] **Base physical defense**
- [x] **Physical defense gain / level**
- [x] **Base magical defense**
- [x] **Magical defense gain / level**
- [x] **Base attack range (in tile)**
- [x] **Attack range gain / level**

(Walk speed (on different terrains, 0 speed on one means it cant pass that terrain) -> Done)
- [x] **Walk speed gain per level** (% over base, should never go over 1.3 of base speed)
- [x] **Attack warmup.** (The time before a creature can trigger its attack.)
- [x] **HP Heal rate per turn** (if not drowsy, hungry)
- [x] **Awakeness lost per turn** (if not sleeping)
- [x] **Hunger growth per turn** (if not sleeping or eating)

**Worker's stats:**

Zero (0) means it can't work on related room/task, All have Base value and Gain per level.
- [x] Dig -> Done.
- [x] Claim -> Done.
- [x] **Dig gain/level**
- [x] **Claim gain/level**

**Comment by akien-mga** [comment]
s/excel/LibreOffice Calc/ ! :-D

**Comment by Bertram25** [comment]
sure. ;)

**Comment by Danimal696** [comment]
i would add saving current weapon on map save; i truly like each creature having different regeneration/hunger/sleep, so some creatures eat like pigs while others barely sleep or dont do at all.

**Comment by Bertram25** [comment]
> i would add saving current weapon on map save;

In fact, this is already done as it is part of the level file for already spawned creatures.

**Comment by Bertram25** [comment]
Also, we should drop the difference seen in the fighter/worker type as it is:
"StrongFighter"/"WeakFighter" --> "Fighter"
"AdvancedWorker"/"BasicWorker" --> "Worker"

While this typification might even be dropped depending on the dance/dig rate the creature has, I do think it shouldn't because the modder should set clearly the type of behaviour the creature should have, especially when speaking of battles.

I mean there will be fighter creatures (especially heroes ones) that can dig but don't flee enemy fighters, like dwarves.

**Comment by hwoarangmy** [comment]
Not to put any pressure but are you far from there ? It's just that I think it is the last step to have something we can call a game. And when we will have that, after a few testing, it would be cool to setup a lan to try all the work done :)

**Comment by Bertram25** [comment]
> Not to put any pressure but are you far from there ?

Arf, sorry, I didn't touch that part yet. I've locally almost fully adapted the GUI to 640x480, but I'm exclusively working on my 0.4.9 assignments. This means I was not planning to work on that before the sound culling and moving source into sub folders, for instance.
But we can always revise the priority of everyone if it helps focusing on releasing. :)
Do you want me to hop on that now, or are you willing to take it?

> it would be cool to setup a lan to try all the work done :)

Ohhhhhh, yeah!!! :D

**Comment by akien-mga** [comment]
Looking at #9, the creature stats and the AI are the two things left between us and the 0.4.9 release :-) The sound culling doesn't appear on the issue, but maybe you can it too if it's something that you want for 0.4.9?

**Comment by hwoarangmy** [comment]
Well, I don't want you to do anything you don't want or change your priorities. I'm just an impatient child waiting for his game :)
But concerning creature stats, I can help you if you want as I believe it won't be such an easy task. Feel free to ask :)

**Comment by Bertram25** [comment]
Hey :)

> The sound culling doesn't appear on the issue, but maybe you can it too if it's something that you want for 0.4.9?

The sound culling is something I should have thought about in the first place. So I consider the sounds still not working IMHO. That's why I'd like to do that for 0.4.9.

> But concerning creature stats, I can help you if you want as I believe it won't be such an easy task. Feel free to ask :)

The major task is to reformat the current stats into the new format aka #151. Afterwards, it's just to point things out and add missing stats to be read from config. :) I believe anybody out of us 4 can handle it but I'll handle it gladly after my next PR. :)
Then, we can review the whole stats and add what's missing even little per little. :)

**Comment by Danimal696** [comment]
dont push yourself too hard, ok? :)

**Comment by Bertram25** [comment]
> dont push yourself too hard, ok? :)

Thanks, but don't worry. ;) I'm quite busy atm, but so far, it's going fine. :)

**Comment by Bertram25** [comment]
Once my latest PR is merged, I'll start this one. :)

**Comment by hwoarangmy** [comment]
Will you do here the needed rooms / active spots to get creatures ?

**Comment by Bertram25** [comment]
> Will you do here the needed rooms / active spots to get creatures ?

I'll be honest, not in the next one. I'll only deal with creature lacking/changing stats data.
It'll be easier to merge IMHO. We'll see who does what afterwards. :)

**Comment by hwoarangmy** [comment]
> I'll be honest, not in the next one

No problem. When possible, I think it is better to have not too big PR (I know I don't always follow that but it is not always easy to split a dev)

**Comment by Danimal696** [comment]
crazy :D
